### PR TITLE
restrict trade aggregations resolution

### PR DESCRIPTION
**Pull request by tomerweller**
to: 1 minute, 15 minutes, 1 hour, 1 day, 1 week

### CODE REVIEW

**Comment by tomerweller** [comment]
in preparation for future caching of these results.
It's worth discussing if these are indeed the right resolutions to restrict to.

File: services/horizon/internal/db2/history/trade_aggregation.go [hunk 1]
```diff
@@ -5,10 +5,19 @@ import (
 
 	sq "github.com/Masterminds/squirrel"
 	"github.com/stellar/go/services/horizon/internal/db2"
+	"github.com/stellar/go/support/errors"
 	. "github.com/stellar/go/support/time"
 	"github.com/stellar/go/xdr"
 )
 
+var AllowedResolutions = map[int64]struct{}{
+	60000:     {}, //1 minute
```
Comment (nullstyle) [inline services/horizon/internal/db2/history/trade_aggregation.go]: How about we use `time.Duration` keys here?

```golang

var AllowedResolutions = map[time.Duration]struct{}{
	time.Minute:        {},
	15 * time.Minute:   {},
	time.Hour:          {},
	24 * time.Hour:     {},
	7 * 24 * time.Hour: {},
}
```

Comment (tomerweller) [reply services/horizon/internal/db2/history/trade_aggregation.go]: I love it. makes me wish I would have used `time.Duration` across all aggregation stuff. Too big of a change for this scope though.

**Comment by nullstyle** [comment]
Tests are failing, fyi

**Comment by nullstyle** [comment]
👍

**Pull request merged**